testing/cloud.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2015, Google, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Common testing tools for cloud samples.
+"""
+
+import inspect
+import os
+import unittest
+
+from .utils import silence_requests
+
+
+GLOBAL_RESOURCE_PATH = os.path.join(
+    os.path.abspath(os.path.dirname(__file__)), 'resources')
+
+
+# Note: these values must also be whitelisted in tox.ini.
+ENVIRONMENT_VARIABLES = frozenset((
+    'GCLOUD_PROJECT',
+    'CLOUD_STORAGE_BUCKET',
+))
+
+
+class Config(object):
+    def __getattr__(self, name):
+        if name not in ENVIRONMENT_VARIABLES:
+            raise AttributeError(
+                'Environment variable {} is not in the whitelist.'.format(
+                    name))
```

Comment:
Attribute error on the next line? I forget Nathaniel's principle here, but I thought in general we avoid dangling lines with only a little bit on it, preferring instead to push down bigger parts sooner.